tests/apps/stateapp/app.go

```diff
@@ -0,0 +1,252 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+const appPort = 3000
+const stateURL = "http://localhost:3500/v1.0/state"
+
+// appState represents a state in this app.
+type appState struct {
+	Data string `json:"data,omitempty"`
+}
+
+// daprState represents a state in Dapr.
+type daprState struct {
+	Key   string    `json:"key,omitempty"`
+	Value *appState `json:"value,omitempty"`
+}
+
+// requestResponse represents a request or response for the APIs in this app.
+type requestResponse struct {
+	StartTime int         `json:"start_time,omitempty"`
+	EndTime   int         `json:"end_time,omitempty"`
+	States    []daprState `json:"states,omitempty"`
+	Message   string      `json:"message,omitempty"`
+}
+
+// indexHandler is the handler for root path
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println("indexHandler is called")
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func save(states []daprState) error {
+	log.Printf("Processing save request for %d entries.", len(states))
+
+	jsonValue, err := json.Marshal(states)
+	if err != nil {
+		log.Printf("Could save states in Dapr: %s", err.Error())
+		return err
+	}
+
+	log.Printf("Posting state to %s with '%s'", stateURL, jsonValue)
+	res, err := http.Post(stateURL, "application/json", bytes.NewBuffer(jsonValue))
+	if err != nil {
+		return err
+	}
+
+	defer res.Body.Close()
+	return nil
+}
+
+func get(key string) (*appState, error) {
+	log.Printf("Processing get request for %s.", key)
+	url, err := createStateURL(key)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("Fetching state from %s", url)
+	// url is created from user input, it is OK since this is a test app only and will not run in prod.
+	/* #nosec */
+	res, err := http.Get(url)
```

Comment:
Can we open the issue to create dapr http api request helpers which can be shared in all test apps? it would be nice to reduce the mistake and make the code cleaner..

Reply:
https://github.com/dapr/dapr/issues/853

Reply:
Done: https://github.com/dapr/dapr/issues/853